frontend/src/features/admin-form/responses/ResponsesPage/storage/useDecryptionWorkers.ts

```diff
@@ -191,95 +195,99 @@ const useDecryptionWorkers = ({
                 }
                 return decryptResult
               })
+              // Step 3: Update the decryption count
               .then((decryptResult) => {
                 decryptionProgress += 1
-                onProgress(decryptionProgress)
```

Comment:
description: do not update modal status until the file is saved on user's device